lib/galaxy/tools/parameters/grouping.py

```diff
@@ -279,10 +285,18 @@ def value_from_basic(self, value, app, ignore_errors=False):
             rval.append(rval_dict)
         return rval
 
+    def get_file_count(self, trans, context):
+        file_count = context.get("file_count", "auto")
+        if file_count == "auto":
```

Comment:
Just a minor one. Can we make it one liner?

`file_count = len(self.get_datatype(trans, context).writable_files) if file_count == "auto" else int(file_count)`

thank you!

Reply:
I'll rebase with this change, thanks!